src/PowerShell/Microsoft.WinGet.Client.Engine/Helpers/AppxModuleHelper.cs

```diff
@@ -210,5 +215,64 @@ private void InstallUiXaml()
                 throw new PSNotImplementedException(Resources.MicrosoftUIXaml27Message);
             }
         }
+
+        private void AddAppxPackageAsUri(string packageUri)
+        {
+            var errorRecord = this.RunScript(string.Format(AddAppxPackageFormat, packageUri));
+
+            if (errorRecord != null)
+            {
+                if (errorRecord.CategoryInfo.Category == ErrorCategory.OpenError)
+                {
+                    this.psCmdlet.WriteDebug($"Failed adding package {packageUri}. Retrying downloading it.");

```

Comment:
NIT: `Failed to add package with uri [{packageUri}]. Retry downloading the package and adding it.`